snuba/datasets/querylog_processor.py

```diff
@@ -0,0 +1,96 @@
+import simplejson as json
+from typing import Any, Mapping, Optional, Sequence, Union
+import uuid
+
+from snuba.processor import MessageProcessor, ProcessedMessage, ProcessorAction
+
+
+class QuerylogProcessor(MessageProcessor):
+    def __to_json_string(self, map: Mapping[str, Any]) -> str:
+        return json.dumps({k: v for k, v in sorted(map.items())})
+
+    def __get_sample(self, sample: Union[int, float]) -> float:
+        """
+        Since sample can be both integer or float between 0 and 1, we just cast
+        int to float so we have a common data type.
+        """
+        if isinstance(sample, int):
+            return float(sample)
+
+        return sample
+
+    def __extract_query_list(
+        self, query_list: Sequence[Mapping[str, Any]]
+    ) -> Mapping[str, Any]:
+        sql = []
+        status = []
+        trace_id = []
+        duration_ms = []
+        stats = []
+        final = []
+        cache_hit = []
+        sample = []
+        max_threads = []
+        num_days = []
+        clickhouse_table = []
+        query_id = []
+        is_duplicate = []
+        consistent = []
+
+        for query in query_list:
+            sql.append(query["sql"])
+            status.append(query["status"])
+            trace_id.append(str(uuid.UUID(query["trace_id"])))
+            # TODO: Calculate subquery duration, for now just insert 0s
+            duration_ms.append(0)
+            stats.append(self.__to_json_string(query["stats"]))
+            final.append(int(query["stats"].get("final", 0)))
+            cache_hit.append(int(query["stats"].get("cache_hit", 0)))
+            sample.append(query["stats"].get("sample", 0))
+            max_threads.append(query["stats"].get("max_threads", 0))
+            num_days.append(query["stats"].get("num_days", 0))
```

Comment:
Mind verifying that these fields in stats are never None ? Because the default value `0` would be applied only if the keys are not present in the dictionary. If the key is present and None, this will try to add NULL to a non nullable column.

Actually applies to all the fields.

Reply:
Good point, I updated to make sure `0` gets set even if the key is there and is none.